File: packages/kt-client/src/main/kotlin/com/google/firebase/auth/OAuthProvider.kt
package com.google.firebase.auth

object OAuthProvider {
    @JvmStatic
    @JvmOverloads
    fun getCredential(
        providerId: String,
        idToken: String?,
        accessToken: String? = null,
        rawNonce: String? = null
    ): AuthCredential = OAuthCredential(providerId, idToken, accessToken, rawNonce)
}

File: packages/swift-client/Sources/FirebaseAuth/OAuthProvider.swift
import Foundation

public enum OAuthProvider {
    public static func credential(
        withProviderID providerID: String,
        idToken: String,
        rawNonce: String? = nil,
        accessToken: String? = nil
    ) -> OAuthCredential {
        OAuthCredential(providerID: providerID, idToken: idToken, rawNonce: rawNonce, accessToken: accessToken)
    }
}
